tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -486,4 +486,35 @@ public function testEnsureDefaultsAreUnescapedFromSchemaIntrospection() : void
         $onlineTable = $this->schemaManager->listTableDetails('test_column_defaults_with_create');
         self::assertSame($default, $onlineTable->getColumn('col1')->getDefault());
     }
+
+    public function testEnsureTableOptionsAreReflectedInMetadata() : void
+    {
+        $this->connection->query('DROP TABLE IF EXISTS test_table_metadata');
+
+        $sql = <<<'SQL'
+CREATE TABLE test_table_metadata(
+  col1 INT NOT NULL AUTO_INCREMENT PRIMARY KEY
+)
+CHARSET utf8
+COLLATE utf8_general_ci
+ENGINE InnoDB
```

Comment:
Hum, authorized creation options could depend on default table engine. That's why I added the engine in a first place. I do not know whether it could make test flacky over configuration or database default engine change.

Reply:
That's fine if the engine is required.